src/satellite/process.ts

```diff
@@ -489,26 +717,39 @@ export class SatelliteProcess implements Satellite {
           incomingChanges.optype
         )
 
-        Object.assign(incomingChanges, { changes, optype })
+        /*
+        console.log('merge_op_tag')
+        console.log(localChanges)
+        console.log(incomingChanges)
+         */
+        const clear_tags = mergeOpTags(localChanges, incomingChanges)
+
+        Object.assign(incomingChanges, { changes, optype, clear_tags })
       }
     }
 
     return incomingTableChanges
   }
 
   async _applyTransaction(transaction: Transaction) {
-    const opLogEntries = fromTransaction(transaction, this.relations)
-
-    await this._apply(opLogEntries, transaction.lsn)
-    this._notifyChanges(opLogEntries)
+    const origin = transaction.origin!
+    if (origin == this._authState!.clientId) {
+      await this._garbageCollectOplog(transaction)
```

Comment:
Can you please elaborate a little on why this is valid? I'm not sure I follow logic of running a bulk delete operation when applying a transaction, and also limiting the delete operation to the transactions in the same microsecond.

Reply:
I should probably add a comment for that in the code but current logic is - to clean oplog when we receive transaction from Electric that corresponds to this part of oplog. Previously we would do such a GC when receiving ack.

Reply:
I'd really appreciate a comment in code about it with this, and probably an explanation as to why is exact equality of items in the oplog on a millisecond time precision is a good indicator of garbage, for posterity